lib/twilio_api_pagination.rb

```diff
@@ -0,0 +1,78 @@
+class TwilioAPIPagination
+  DEFAULT_LIMIT_PER_PAGE = 50
+
+  attr_reader :paginated_collection, :resources, :requested_uri, :params
+
+  def initialize(resources, requested_uri)
+    @resources = resources
+    @requested_uri = URI.parse(requested_uri)
+    @params = parse_params
+    @paginated_collection = paginate_resources
+  end
+
+  def info
+    @info ||= {
+      uri: requested_uri,
+      page: page_number,
+      page_size: page_size,
+      first_page_uri: pagination_link(PageToken: nil, Page: 0),
+      next_page_uri: next_cursor && pagination_link(PageToken: "PA#{next_cursor}", Page: page_number + 1),
+      previous_page_uri: prev_cursor && pagination_link(PageToken: "PB#{prev_cursor}", Page: (page_number.zero? ? 0 : page_number - 1))
```

Comment:
`[page_number - 1, 0].max` might be easier to understand

Reply:
I wrote exactly at first, then I thought you might said it was hard to understand, so that's why I use the ternary operator, so it's not fancy with ruby trick and our head will translate it again. It's okay we could switch to use the `max` method.